dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/IsinDeltaBuilder.kt

```diff
@@ -0,0 +1,116 @@
+) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Creates the delta file of the updated LEI-ISIN mapping file
+     * @param newMappingFile latest version of the LEI-ISIN mapping file
+     * @param oldMappingFile previously stored version of the LEI-ISIN mapping file, if one exists
+     */
+    fun createDeltaOfMappingFile(newMappingFile: File, oldMappingFile: File?): Map<String, String> {
+        val newMapping = parseCsvToGroupedMap(newMappingFile)
+        if (oldMappingFile == null) {
+            replaceOldMappingFile(newMappingFile)
+            return newMapping
+        }
+        val oldMapping = parseCsvToGroupedMap(oldMappingFile)
+        val deltaMapping = findLeisWithUpdatedIsin(newMapping, oldMapping)
+        replaceOldMappingFile(newMappingFile)
+        return deltaMapping
+    }
+
+    /**
+     * Compares the two LEI-ISIN maps and identifies LEIs that have new ISINs
+     * @param newMap the new LEI-ISIN map
+     * @param oldMap the old LEI-ISIN map
+     * @return map of changed LEI-ISINs
+     */
+    fun findLeisWithUpdatedIsin(newMap: Map<String, String>, oldMap: Map<String, String>): Map<String, String> {
+        val deltaMap = mutableMapOf<String, String>()
+
+        for ((lei, newIsins) in newMap) {
+            val oldIsins = oldMap[lei]
+
+            if (oldIsins == null || oldIsins != newIsins) {
+                deltaMap[lei] = newIsins
+            }
+        }
+        return deltaMap
+    }
+
+    /**
+     * Coverts CSV file to a LEI-ISIN map, while also aggregating all ISINs of a specific
+     * LEI into one comma-separated string
+     * @param csvFile the file to be parsed
+     * @return map of LEI-ISINs
+     */
+    private fun parseCsvToGroupedMap(csvFile: File): Map<String, String> {
+        val csvMapper = CsvMapper()
+        csvMapper.registerModule(kotlinModule())
+
+        val csvSchema = CsvSchema.builder()
+            .addColumn("LEI")
+            .addColumn("ISIN")
+            .setUseHeader(true)
+            .build()
+
+        val mappings = mutableMapOf<String, StringBuilder>()
+
+        try {
+            val csvParser = csvMapper
+                .readerFor(Map::class.java)
+                .with(csvSchema)
+                .readValues<Map<String, String>>(csvFile)
+
+            csvParser.forEach { entry ->
+                val lei = entry["LEI"]
+                val isin = entry["ISIN"]
+
+                if (lei != null && isin != null && mappings.containsKey(lei)) {
+                    mappings[lei]?.append(",")
+                    mappings[lei]?.append(isin)
+                } else if (lei != null && isin != null && !mappings.containsKey(lei)) {
+                    mappings[lei] = StringBuilder(isin)
+                }
+            }
+        } catch (e: FileSystemException) {
+            logger.error("Error while parsing CSV: ${e.message}")
```

Comment:
Why catch that error? Throw it. This should not happen in a production scenario.